cmd/main.go

```diff
@@ -122,7 +136,43 @@ func awaitStopSignal(stopChan chan struct{}) {
 	close(stopChan)
 }
 
+func awaitLeaderDeposed(leaderContext context.Context) {
+	// If the leader Context is finished, that's because we stopped leading.
+	// so we will crash.
+	<-leaderContext.Done()
+
+	log.Fatal("Was deposed as leader, exiting.")
+
+}
+
+// startLeaderElection creates and starts the leader election
+func startLeaderElection(client kubernetes.Interface, config k8s.LeaderElectConfig) (context.Context, error) {
+	eventsScheme := runtime.NewScheme()
+	if err := coreV1.AddToScheme(eventsScheme); err != nil {
+		return nil, err
+	}
+
+	// Start events recorder
+	eventBroadcaster := record.NewBroadcaster()
+	recorder := eventBroadcaster.NewRecorder(eventsScheme, coreV1.EventSource{Component: "escalator"})
```

Comment:
it seems that the recorder won't work properly unless you initialise it, some related blog post: https://kubernetes.io/blog/2018/01/reporting-errors-using-kubernetes-events/